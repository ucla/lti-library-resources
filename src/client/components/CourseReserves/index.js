import React, { useState, useEffect } from 'react';
import axios from 'axios';
import Iframe from 'react-iframe';
import PropTypes from 'prop-types';
import { Alert } from '@instructure/ui-alerts';
import { Button } from '@instructure/ui-buttons';
import axiosRetry from 'axios-retry';
import { getLtik } from '../../services/ltik';

axiosRetry(axios);

const CourseReserves = ({ context, setError }) => {
  // Holds URL of course reserve, or '' if there is none
  const [url, setUrl] = useState('');

  // Called once, to retrieve course reserve URL from database
  const getUrl = () => {
    if (typeof context.context === 'undefined') {
      return;
    }

    const ltik = getLtik();
    axios
      .get(`${process.env.LTI_APPROUTE}/api/getreserveurl?ltik=${ltik}`, {
        params: {
          shortname: context.context.label,
        },
      })
      .then((res) => {
        setUrl(res.data.reserve);
        setError(null);
      })
      .catch((err) => {
        setError({
          err,
          msg: 'Something went wrong when retrieving course reserve...',
        });
      });
  };

  // Get URL (only load once)
  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(getUrl, []);

  return (
    <div>
      {url ? (
        <Iframe url={url} />
      ) : (
        <div>
          <Alert variant="warning" margin="small">
            There are no library reserves for this course.
          </Alert>
          <div>
            <Button
              color="primary"
              margin="small"
              href="http://www.library.ucla.edu/use/borrow-renew-return/course-reserves/information-instructors"
              target="_blank"
              rel="noopener roreferrer"
            >
              Request library reserves
            </Button>
            <Button
              color="secondary"
              margin="small"
              href="http://ucla.libsurveys.com/rg-feedback"
              target="_blank"
              rel="noopener roreferrer"
            >
              Library resources feedback
            </Button>
          </div>
        </div>
      )}
    </div>
  );
};

CourseReserves.propTypes = {
  context: PropTypes.object.isRequired,
  setError: PropTypes.func,
};

export default CourseReserves;
