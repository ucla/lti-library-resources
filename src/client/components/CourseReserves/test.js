import React from 'react';
import CourseReserves from '.';
import { shallow } from 'enzyme';

it('Correctly displays iframe when URL exists', () => {
  const wrapper = shallow(
    <CourseReserves url={"test"} />,
  );
  expect(wrapper).toMatchSnapshot();
});

it('Correctly does not display iframe when URL is empty', () => {
  const wrapper = shallow(
    <CourseReserves url={""} />,
  );
  expect(wrapper).toMatchSnapshot();
});

it('Correctly does not display iframe when URL is none', () => {
  const wrapper = shallow(
    <CourseReserves />,
  );
  expect(wrapper).toMatchSnapshot();
});