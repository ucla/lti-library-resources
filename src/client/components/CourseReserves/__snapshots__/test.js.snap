// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly displays iframe when URL exists 1`] = `
<div>
  <Iframe
    url="test"
  />
</div>
`;

exports[`Correctly does not display iframe when URL is empty 1`] = `
<div>
  <Iframe
    url=""
  />
</div>
`;

exports[`Correctly does not display iframe when URL is none 1`] = `
<div>
  <Iframe />
</div>
`;
